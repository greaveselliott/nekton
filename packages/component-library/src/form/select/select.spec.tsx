import React from 'react';
import { shallow } from 'enzyme';
import Select from './select.component';

const selectFormOptions = [
  {
    value: 'option1',
    text: 'Option One'
  },
  {
    value: 'option2',
    text: 'Option Two'
  },
  {
    value: 'option3',
    text: 'Option Three'
  }
];

describe('Select', () => {
  it('renders a select element with placeholder text', () => {
    expect(
      shallow(
        <Select
          placeholderText="placeholder"
          selectOptions={selectFormOptions}
        />
      )
    ).toMatchSnapshot();
  });

  it('can render a disabled select element', () => {
    expect(
      shallow(
        <Select
          disabled
          placeholderText="placeholder"
          selectOptions={selectFormOptions}
        />
      )
    ).toMatchSnapshot();
  });

  it('can render a select element', () => {
    expect(
      shallow(<Select name="test-select" selectOptions={selectFormOptions} />)
    ).toMatchSnapshot();
  });
});
