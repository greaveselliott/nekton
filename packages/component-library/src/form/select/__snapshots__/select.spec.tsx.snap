// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select can render a disabled select element 1`] = `
<div
  className="select-wrapper"
>
  <select
    className="select-field"
    disabled={true}
  >
    <option
      value=""
    >
      placeholder
    </option>
    <option
      key="option1"
      value="option1"
    >
      Option One
    </option>
    <option
      key="option2"
      value="option2"
    >
      Option Two
    </option>
    <option
      key="option3"
      value="option3"
    >
      Option Three
    </option>
  </select>
</div>
`;

exports[`Select can render a select element 1`] = `
<div
  className="select-wrapper"
>
  <select
    className="select-field"
    name="test-select"
  >
    <option
      key="option1"
      value="option1"
    >
      Option One
    </option>
    <option
      key="option2"
      value="option2"
    >
      Option Two
    </option>
    <option
      key="option3"
      value="option3"
    >
      Option Three
    </option>
  </select>
</div>
`;

exports[`Select renders a select element with placeholder text 1`] = `
<div
  className="select-wrapper"
>
  <select
    className="select-field"
  >
    <option
      value=""
    >
      placeholder
    </option>
    <option
      key="option1"
      value="option1"
    >
      Option One
    </option>
    <option
      key="option2"
      value="option2"
    >
      Option Two
    </option>
    <option
      key="option3"
      value="option3"
    >
      Option Three
    </option>
  </select>
</div>
`;
